### PR TITLE
Update misc methods to where syntax and use Vector instead of Array where it makes sense

### PR DESCRIPTION
**Pull request by musm**
continuation of https://github.com/JuliaLang/julia/pull/21428

### CODE REVIEW

**Review by andreasnoack — APPROVED** [review]

File: base/reducedim.jl [hunk 1]
```diff
@@ -688,7 +688,7 @@ julia> findmax(A,2)
 ([2; 4], [3; 4])
 ```
 """
-function findmax{T}(A::AbstractArray{T}, region)
+function findmax(A::AbstractArray{T}, region) where {T}
```
Comment (musm) [inline base/reducedim.jl]: Reminder to remove the braces

**Comment by musm** [comment]
cc @Sacha0  (I hope you can take review this PR, a lot of changes are in the sparse codes, thanks)
These changes should also help reduce compile times

**Comment by tkelman** [comment]
State as of https://github.com/JuliaLang/julia/pull/21457/commits/334845735f77d9ace0b5be2fe208310972ebe63c lgtm, done here? Should be squashed on merge.

**Comment by musm** [comment]
Yes I'm done, will open new prs for other stuff, thanks

**Review by Sacha0 — COMMENTED** [review]
late-to-the-party lgtm! :)

**Pull request merged**